include/game/bases/d_ac_py_key.hpp

```diff
@@ -0,0 +1,107 @@
+    int buttonUp() const; ///< Returns whether the up button is pressed down.
+    int buttonDown() const; ///< Returns whether the down button is pressed down.
+    int buttonLeft() const; ///< Returns whether the left button is pressed down.
+    int buttonRight() const; ///< Returns whether the right button is pressed down.
+    int triggerUp() const; ///< Returns whether the up button was pressed this frame.
+    int triggerDown() const; ///< Returns whether the down button was pressed this frame.
+    int triggerLeft() const; ///< Returns whether the left button was pressed this frame.
+    int triggerRight() const; ///< Returns whether the right button was pressed this frame.
+    int triggerTwo() const; ///< Returns whether the two button was pressed this frame.
+    int buttonTwo() const; ///< Returns whether the two button is pressed down.
+    int triggerOne() const; ///< Returns whether the one button was pressed this frame.
+    int buttonOne() const; ///< Returns whether the one button is pressed down.
+    int buttonDush() const; ///< Returns whether the run button is pressed down.
+    int triggerAttack() const; ///< Returns whether the attack button (1 / B) was pressed this frame.
+    bool triggerEat(); ///< Returns whether the eat button (1 / B) was pressed this frame.
+    int buttonCarry() const; ///< Returns whether the carry button (1 / B) is pressed down.
+    bool triggerFire(); ///< Returns whether the fire button was pressed this frame.
+    int triggerJumpBase() const; ///< Returns whether the jump button (2) was pressed this frame.
+    bool triggerJump(); ///< Returns whether the jump button (2) was pressed this frame.
+    int buttonJump(); ///< Returns whether the jump button (2) is pressed down.
+    int buttonYoshiJump() const; ///< Returns whether the Yoshi jump button (2) is pressed down.
+    u8 triggerShakeJump() const; ///< Returns whether a shake event was triggered this frame.
+    void clearShakeJump(); ///< Clears the shake event state.
+    bool buttonWalk(int *direction) const; ///< Returns whether left / right is pressed down. Sets the direction to 0 for right and 1 for left.
+    int buttonCrouch() const; ///< Returns whether the crouch button (down) is pressed down.
+    int buttonHipAttack() const; ///< Returns whether the correct buttons (down, no other directions) for a hip attack is pressed down.
+    int buttonDoor() const; ///< Returns whether the correct buttons (up, no left or right) for a door open is pressed down.
+    bool triggerJumpBuf(int n); ///< Returns whether a jump occurred in the last n frames.
+    void onDemoButton(int); ///< Sets a button for demo mode to pressed down.
+    void offDemoButton(int); ///< Sets a button for demo mode to not pressed down.
+    void onDemoTrigger(int); ///< Sets a button for demo mode to pressed this frame.
+    void onDemoShake(); ///< Sets a shake event for demo mode to be triggered this frame.
```

Comment:
I know this is pedantic, but maybe add some whitespace (such as grouping by function type) instead of having this massive shopping list.